clang-tools-extra/clang-doc/tool/ClangDocMain.cpp

```diff
@@ -247,6 +247,22 @@ sortUsrToInfo(llvm::StringMap<std::unique_ptr<doc::Info>> &USRToInfo) {
   }
 }
 
+llvm::Error runMappingPhase(tooling::ToolExecutor *Executor,
+  clang::doc::ClangDocContext &CDCtx, tooling::ArgumentsAdjuster &ArgAdjuster, bool IgnoreMappingFailures ) {
```

Comment:
I was thinking more of an API that took in the Error from the executor and returned something usable with ExitOnErr.

```cpp
ExitOnErr(helper_function(Executor->execute(doc::newMapperActionFactory(CDCtx), ArgAdjuster));
```